code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -2175,3 +2180,159 @@ All effects don't start immediately, but rather get worse over time; the rate is
 			M.adjustFireLoss(-16, 0)
 
 	..()
+
+/datum/reagent/consumable/ethanol/plasmaflood/on_mob_end_metabolize(mob/living/L)
+	to_chat(L, "<span class='warning'>You no longer feel immune to burning!</span>")
+	. = ..()
+
+/datum/reagent/consumable/ethanol/fourthwall
+	name = "Fourth Wall"
+	description = "This substance seems like it shouldn't exist."
+	color = "#0b43a3"
+	boozepwr = 0 //I feel like brain traumas is enough
+	quality = DRINK_GOOD
+	metabolization_rate = 0.3
+	taste_description = "binary"
+	glass_icon_state = "fourthwallglass"
+	glass_name = "Fourth Wall"
+	glass_desc = "Just looking at this makes your head hurt."
+	var/trauma_list[0]
+
+/datum/reagent/consumable/ethanol/fourthwall/proc/traumaweightpick(var/mild,var/severe,var/special)
+	return pick(pickweight(list(subtypesof(/datum/brain_trauma/mild) = mild, subtypesof(/datum/brain_trauma/severe) = severe, subtypesof(/datum/brain_trauma/special) = special)))
+
+/datum/reagent/consumable/ethanol/fourthwall/on_mob_metabolize(mob/living/carbon/M)
+	to_chat(M, "<span class='warning'>Your mind breaks, as you realize your reality is just some comupter game.</span>")
+	var/trauma = traumaweightpick(60,40,0)
+	trauma_list += trauma
+	M.gain_trauma(trauma, TRAUMA_RESILIENCE_ABSOLUTE)
```

Comment:
I'd add a check to make sure they dont have the trauma already, otherwise you can drink the drink, get a trauma you already had, and then get it removed by ending the drink.

Reply:
Code for fourth wall was improved, and this already happens in the gain_trauma proc